In [1]:
import pandas as pd

print("Notebook is working!")


Notebook is working!


In [3]:
import os
print(os.listdir("../data/raw"))


['olist_customers_dataset.csv', 'olist_geolocation_sample.csv', 'olist_orders_sample.csv', 'olist_order_items_sample.csv', 'olist_order_payments_sample.csv', 'olist_order_reviews_sample.csv', 'olist_products_sample.csv', 'olist_sellers_sample.csv', 'product_category_name_translation.csv']


In [4]:
import os

print(os.listdir("../data/raw"))


['olist_customers_dataset.csv', 'olist_customers_dataset_sample.csv', 'olist_geolocation_dataset.csv', 'olist_geolocation_sample.csv', 'olist_orders_dataset.csv', 'olist_orders_sample.csv', 'olist_order_items_dataset.csv', 'olist_order_items_sample.csv', 'olist_order_payments_dataset.csv', 'olist_order_payments_sample.csv', 'olist_order_reviews_dataset.csv', 'olist_order_reviews_sample.csv', 'olist_products_dataset.csv', 'olist_products_sample.csv', 'olist_sellers_dataset.csv', 'olist_sellers_sample.csv', 'product_category_name_translation.csv', 'product_category_name_translation_sample.csv']


In [5]:
import pandas as pd

# Load datasets from raw folder
orders      = pd.read_csv("../data/raw/olist_orders_dataset.csv")
customers   = pd.read_csv("../data/raw/olist_customers_dataset.csv")
order_items = pd.read_csv("../data/raw/olist_order_items_dataset.csv")
payments    = pd.read_csv("../data/raw/olist_order_payments_dataset.csv")
products    = pd.read_csv("../data/raw/olist_products_dataset.csv")
sellers     = pd.read_csv("../data/raw/olist_sellers_dataset.csv")
reviews     = pd.read_csv("../data/raw/olist_order_reviews_dataset.csv")
geolocation = pd.read_csv("../data/raw/olist_geolocation_dataset.csv")
categories  = pd.read_csv("../data/raw/product_category_name_translation.csv")

# Quick check
print("Orders:", orders.shape)
print("Customers:", customers.shape)
print("Order Items:", order_items.shape)
print("Payments:", payments.shape)


Orders: (99441, 8)
Customers: (99441, 5)
Order Items: (112650, 7)
Payments: (103886, 5)


In [6]:
def explore_dataset(df, name):
    print(f"--- {name} ---")
    print(f"Shape: {df.shape}")
    print("\nMissing values:")
    print(df.isna().sum())
    print("\nDtypes:")
    print(df.dtypes)
    print("\nDuplicates:", df.duplicated().sum())
    print("\nPreview:")
    display(df.head())
    print("\n" + "="*80 + "\n")


In [7]:
datasets = {
    "Orders": orders,
    "Customers": customers,
    "Order Items": order_items,
    "Payments": payments,
    "Products": products,
    "Sellers": sellers,
    "Reviews": reviews,
    "Geolocation": geolocation,
    "Categories": categories
}

for name, df in datasets.items():
    explore_dataset(df, name)


--- Orders ---
Shape: (99441, 8)

Missing values:
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

Dtypes:
order_id                         object
customer_id                      object
order_status                     object
order_purchase_timestamp         object
order_approved_at                object
order_delivered_carrier_date     object
order_delivered_customer_date    object
order_estimated_delivery_date    object
dtype: object

Duplicates: 0

Preview:


order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00



--- Customers ---
Shape: (99441, 5)

Missing values:
customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

Dtypes:
customer_id                 object
customer_unique_id          object
customer_zip_code_prefix     int64
customer_city               object
customer_state              object
dtype: object

Duplicates: 0

Preview:


customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP



--- Order Items ---
Shape: (112650, 7)

Missing values:
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

Dtypes:
order_id                object
order_item_id            int64
product_id              object
seller_id               object
shipping_limit_date     object
price                  float64
freight_value          float64
dtype: object

Duplicates: 0

Preview:


order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14



--- Payments ---
Shape: (103886, 5)

Missing values:
order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

Dtypes:
order_id                 object
payment_sequential        int64
payment_type             object
payment_installments      int64
payment_value           float64
dtype: object

Duplicates: 0

Preview:


order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45



--- Products ---
Shape: (32951, 9)

Missing values:
product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

Dtypes:
product_id                     object
product_category_name          object
product_name_lenght           float64
product_description_lenght    float64
product_photos_qty            float64
product_weight_g              float64
product_length_cm             float64
product_height_cm             float64
product_width_cm              float64
dtype: object

Duplicates: 0

Preview:


product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0



--- Sellers ---
Shape: (3095, 4)

Missing values:
seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

Dtypes:
seller_id                 object
seller_zip_code_prefix     int64
seller_city               object
seller_state              object
dtype: object

Duplicates: 0

Preview:


seller_id  seller_zip_code_prefix  \
0  3442f8959a84dea7ee197c632cb2df15                   13023   
1  d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2  ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3  c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4  51a04a8a6bdcb23deccc82b0b80742cf                   12914   

         seller_city seller_state  
0           campinas           SP  
1         mogi guacu           SP  
2     rio de janeiro           RJ  
3          sao paulo           SP  
4  braganca paulista           SP



--- Reviews ---
Shape: (99224, 7)

Missing values:
review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

Dtypes:
review_id                  object
order_id                   object
review_score                int64
review_comment_title       object
review_comment_message     object
review_creation_date       object
review_answer_timestamp    object
dtype: object

Duplicates: 0

Preview:


review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53



--- Geolocation ---
Shape: (1000163, 5)

Missing values:
geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

Dtypes:
geolocation_zip_code_prefix      int64
geolocation_lat                float64
geolocation_lng                float64
geolocation_city                object
geolocation_state               object
dtype: object

Duplicates: 261831

Preview:


geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                         1037       -23.545621       -46.639292   
1                         1046       -23.546081       -46.644820   
2                         1046       -23.546129       -46.642951   
3                         1041       -23.544392       -46.639499   
4                         1035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP



--- Categories ---
Shape: (71, 2)

Missing values:
product_category_name            0
product_category_name_english    0
dtype: int64

Dtypes:
product_category_name            object
product_category_name_english    object
dtype: object

Duplicates: 0

Preview:


product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

# Olist Dataset Overview

This notebook provides a high-level exploration of the raw Olist e-commerce datasets.  
We examine structure, missing values, and potential join keys.

---

## 1. Orders Dataset (`olist_orders_dataset.csv`)
- Rows: 99.441, Columns: 8
- Missing values: `order_approved_at`, `order_delivered_carrier_date`, `order_delivered_customer_date` have nulls
- Duplicates: 0
- Key columns: `order_id`, `customer_id`
- Notes: Timestamp columns need conversion to datetime.

---

## 2. Customers Dataset (`olist_customers_dataset.csv`)
- Rows: 99,441, Columns: 5
- Missing values: None
- Duplicates: `customer_unique_id` repeats (same customer can place multiple orders)  
- Key columns: `customer_id`, `customer_unique_id`
- Notes: Each `customer_id` is unique per order

---

## 3. Order Items Dataset (`olist_order_items_dataset.csv`)
- Rows: 112,650, Columns: 7
- Missing values: None
- Duplicates: No exact duplicates, but multiple rows per `order_id` when an order has multiple products.
- Key columns: `order_id`, `product_id`, `seller_id`
- Notes: (e.g. multiple items per order)

---

## 4. Payments Dataset (`olist_order_payments_dataset.csv`)
- Rows: 103,886, Columns: 5
- Missing values: None
- Duplicates: No exact duplicates, but some orders appear multiple times (split payments).
- Key columns: `order_id`
- Notes: (e.g. multiple payments per order possible)

---

## 5. Products Dataset (`olist_products_dataset.csv`)
- Rows: 32,951, Columns: 9
- Missing values:`product_category_name` → ~6,000 nulls , `product_name_length`, `product_description_length`, `product_photos_qty` → many nulls 
- Duplicates: None
- Key columns: `product_id`
- Notes: (e.g. translation available in categories table)

---

## 6. Sellers Dataset (`olist_sellers_dataset.csv`)
- Rows: 3,100, Columns: 4
- Missing values: None
- Duplicates: None
- Key columns: `seller_id`
- Notes: (e.g. seller locations can be joined with geolocation)

---

## 7. Reviews Dataset (`olist_order_reviews_dataset.csv`)
- Rows: 100,000, Columns: 7
- Missing values:`review_comment_title` → many nulls , `review_comment_message` → many nulls  
- Duplicates: - Duplicates: Some orders may have multiple reviews. 
- Key columns: `review_id`, `order_id`
- Notes: (e.g. multiple reviews per order in rare cases)

---

## 8. Geolocation Dataset (`olist_geolocation_dataset.csv`)
- Rows: 1,000,163, Columns: 5
- Missing values: None
- Duplicates: Many → same zip code prefix can map to multiple coordinates.  
- Key columns: `geolocation_zip_code_prefix`
- Notes: Used for mapping, may require aggregation.

---

## 9. Categories Dataset (`product_category_name_translation.csv`)
- Rows: 71, Columns: 2
- Missing values: None
- Duplicates: None
- Key columns: `product_category_name`
- Notes: (e.g. translation from Portuguese to English)

---

# Summary
- Most datasets are clean and well-structured.  
- Key issues:  
  - Missing timestamps in **Orders**  
  - Missing product attributes in **Products**  
  - Many nulls in **Reviews** text fields  
  - Duplicates in **Geolocation** (expected due to multiple points per zip code).  

Next step → **Data Cleaning (`02_cleaning.ipynb`)** where I’ll:  
1. Handle missing values  
2. Convert data types (dates, numeric)  
3. Remove/handle duplicates  
4. Create cleaned datasets for analysis